aten/src/ATen/ParallelNative.cpp

```diff
@@ -87,15 +86,15 @@ void _run_with_pool(const std::function<void(int, size_t)>& fn, size_t range) {
   // Run the first task on the current thread directly.
   fn(0, 0);
 #else
-  caffe2::ThreadPool* pool = caffe2::mobile_threadpool();
-  if (pool) {
-    // caffe2::ThreadPool can utilize the current thread.
-    pool->run(fn, range);
-  } else {
-    for (size_t i = 0; i < range; ++i) {
-      fn(0, i);
-    }
-  }
+  caffe2::PThreadPool* const pool = caffe2::pthreadpool();
+  TORCH_INTERNAL_ASSERT(pool, "Invalid thread pool!");
+
+  pool->run(
+    // PThreadPool::run() is blocking.  A std::function [const] reference to
```

Comment:
Why not maintain the same function signature as (int, size_t) instead of constructing another wrapper lambda?

Reply:
This code went through several iterations and maybe I would have done it differently if development had followed a different course. :)  Having said that, the main reason I left the first argument out is mainly because it seems to no longer be used at all.  Even this usage of the thread pool has no use for that argument and just hardcodes the argument to 0.  I can just as well move this 0 to PThreadPool.